File: 0xxx/00xx/000x/0009.js
/**
 * @param {number} x
 * @return {boolean}
 */
const isPalindrome = function (x) {
  if (x < 0) return false
  if (x < 10) return true
  const str = String(x)
  const length = str.length
  for (let i = 0; i < length / 2; i++) {
    if (str[i] !== str[length - i - 1]) return false
  }
  return true
}

File: 0xxx/00xx/000x/0009.py
class Solution:
  def isPalindrome(self, x: int) -> bool:
    if x < 0: return False
    if x < 10: return True

    strX = str(x)
    leng = len(strX)
    for i in range(leng // 2):
      if strX[i] != str[leng - i - 1]: return False
    return True